src/daft-core/src/lit/image_buffer.rs

```diff
@@ -0,0 +1,730 @@
+            }
+            DynamicImage::ImageLumaA8(img_buf) => {
+                DaftImageBuffer::<'a>::LA(image_buffer_vec_ref_to_cow(img_buf))
+            }
+            DynamicImage::ImageRgb8(img_buf) => {
+                DaftImageBuffer::<'a>::RGB(image_buffer_vec_ref_to_cow(img_buf))
+            }
+            DynamicImage::ImageRgba8(img_buf) => {
+                DaftImageBuffer::<'a>::RGBA(image_buffer_vec_ref_to_cow(img_buf))
+            }
+            DynamicImage::ImageLuma16(img_buf) => {
+                DaftImageBuffer::<'a>::L16(image_buffer_vec_ref_to_cow(img_buf))
+            }
+            DynamicImage::ImageLumaA16(img_buf) => {
+                DaftImageBuffer::<'a>::LA16(image_buffer_vec_ref_to_cow(img_buf))
+            }
+            DynamicImage::ImageRgb16(img_buf) => {
+                DaftImageBuffer::<'a>::RGB16(image_buffer_vec_ref_to_cow(img_buf))
+            }
+            DynamicImage::ImageRgba16(img_buf) => {
+                DaftImageBuffer::<'a>::RGBA16(image_buffer_vec_ref_to_cow(img_buf))
+            }
+            DynamicImage::ImageRgb32F(img_buf) => {
+                DaftImageBuffer::<'a>::RGB32F(image_buffer_vec_ref_to_cow(img_buf))
+            }
+            DynamicImage::ImageRgba32F(img_buf) => {
+                DaftImageBuffer::<'a>::RGBA32F(image_buffer_vec_ref_to_cow(img_buf))
+            }
+            _ => unimplemented!("{dyn_img:?} not implemented"),
+        }
+    }
+}
+
+impl<'a> From<DaftImageBuffer<'a>> for DynamicImage {
+    fn from(daft_buf: DaftImageBuffer<'a>) -> Self {
+        match daft_buf {
+            DaftImageBuffer::L(buf) => image_buffer_cow_to_vec(buf).into(),
+            DaftImageBuffer::LA(buf) => image_buffer_cow_to_vec(buf).into(),
+            DaftImageBuffer::RGB(buf) => image_buffer_cow_to_vec(buf).into(),
+            DaftImageBuffer::RGBA(buf) => image_buffer_cow_to_vec(buf).into(),
+            DaftImageBuffer::L16(buf) => image_buffer_cow_to_vec(buf).into(),
+            DaftImageBuffer::LA16(buf) => image_buffer_cow_to_vec(buf).into(),
+            DaftImageBuffer::RGB16(buf) => image_buffer_cow_to_vec(buf).into(),
+            DaftImageBuffer::RGBA16(buf) => image_buffer_cow_to_vec(buf).into(),
+            DaftImageBuffer::RGB32F(buf) => image_buffer_cow_to_vec(buf).into(),
+            DaftImageBuffer::RGBA32F(buf) => image_buffer_cow_to_vec(buf).into(),
+        }
+    }
+}
+
+fn convert_img_fmt(fmt: ImageFormat) -> image::ImageFormat {
+    match fmt {
+        ImageFormat::PNG => image::ImageFormat::Png,
+        ImageFormat::JPEG => image::ImageFormat::Jpeg,
+        ImageFormat::TIFF => image::ImageFormat::Tiff,
+        ImageFormat::GIF => image::ImageFormat::Gif,
+        ImageFormat::BMP => image::ImageFormat::Bmp,
+    }
+}
+
+/// Wrapper around image::DynamicImage to implement certain traits
+#[derive(Debug, Clone, PartialEq)]
+pub struct ImageBufferWrapper(pub DynamicImage);
```

Comment:
can we just call this `Image` or something? if this is the type that we use within `Literal` and elsewhere, then the `DynamicImage` is just an implementation detail right? The _Wrapper naming feels unnecessary

Reply:
Good idea. I was doing this to make it similar to the existing `DaftImageBuffer` but what I might also do is rename that to `CowImage`